### ISSUE TITLE
[terra-form-input] Numeric variant not working as expected in mobile

### ISSUE THREAD

**Issue by jeremyfuksa** [cerner/terra-core#3347]
# Bug Report

## Description

Numeric input works as expected on desktop browsers, but on mobile it functions exactly as a text input.

## Steps to Reproduce

1. Visit https://engineering.cerner.com/terra-ui/components/terra-form-input/form-input/input on a mobile device
2. Go to the Numeric Input example
3. Enter a string

## Additional Context / Screenshots

![IMG_3A8FEFB94163-1](https://user-images.githubusercontent.com/726351/107568708-00b04b80-6bad-11eb-8329-959119a21abc.jpeg)

## Expected Behavior

Only numbers should be able to be entered into this input variant.

## Possible Solution

Additional attributes could be added to the variant to force a numeric-only keyboard on mobile. See https://codesandbox.io/embed/ios-force-numeric-keypad-on-input-typenumber-demo-41dkz

## Environment

* Component Name and Version: Terra Form Input v4.11.0
* Browser Name and Version: Microsoft Edge
* Operating System and version (desktop or mobile): iOS 14

**Comment by ryanthemanuel** [comment]
One thing we could do with this is adding the sandbox that @jeremyfuksa linked to as an example (after we validate that it works across mobile browsers like we expect).

**Comment by neilpfeiffer** [comment]
Couple of notes, instructions: 

Terra currently does not have a "numeric input" component, only a few example on the doc page for how to add pass through html native attributes that get spread to the input element, but there is currently no documentation or instructions:
- [..doc/common/NumberInputExample.jsx](https://github.com/cerner/terra-core/blob/main/packages/terra-form-input/src/terra-dev-site/doc/common/NumberInputExample.jsx)
- [..doc/example/uncontrolled/FileExample.jsx](https://github.com/cerner/terra-core/blob/main/packages/terra-form-input/src/terra-dev-site/doc/example/uncontrolled/FileExample.jsx)
- [..doc/example/NumberInputField.jsx](https://github.com/cerner/terra-core/blob/main/packages/terra-form-input/src/terra-dev-site/doc/example/NumberInputField.jsx)

Work Items
1. validate the additional html attributes work for all mobile browsers (and their device keyboards) that Terra supports
2. identify the limitations that will exist using these methods for the individual browser/devices 
a. example: iOS only allows 0-9, no dashes, periods, etc. and no way to change the keyboard/s
b. https://css-tricks.com/finger-friendly-numerical-inputs-with-inputmode/
3. If item 1 and 2 are fine, create a new doc page to collect & move the three examples mentioned above, add another new example with just these new mobile-keyboard-specific helper attributes as a second number input example, and we should use more guide-type walkthrough sections that show and explain how to add these native html attributes to the basic Terra components and explains the prop spreading that will occur. Please use the code examples sections like in [terra-table](https://engineering.cerner.com/terra-ui/components/terra-table/table/guides/single-select-table), and the new [notices doc](https://engineering.cerner.com/terra-ui/dev_tools/cerner/documentation-components/notice) component as needed (yellow/caution for warnings, green/information for accessibility tips and things to remember, blue/ux-recomendation for Design Standards instructions/tips).

**Comment by pranav300** [comment]
What I have found is, that if we make `type="number"` then we cannot limit the no. of characters in the input. And if we make `type="text"` then we can add a limit for the characters, but the users can type in all the characters in the desktop browsers.

I believe for input, we need both `limit the characters and not allow all the characters`. Any thoughts? @neilpfeiffer @ryanthemanuel  @ShettyAkarsh
